File: 1-Layers/1.1-Presentation/RoleWebApi.Client.Web/ClientApp/src/app/models/dtos/client-paged-list.ts
import { ClientDTO } from "./index";

export class ClientPagedList {
  public PageIndex: number;
  public PageSize: number;
  public TotalCount: number;
  public TotalPages: number;
  public IndexFrom: number;
  public Items: ClientDTO[];

  constructor() {
    this.Items = new Array<ClientDTO>();
  }
}

File: 1-Layers/1.5-Infrastructure/CrossCutting/RoleWebApi.Infrastructure.CrossCutting.DTO/ClientPagedList.cs
namespace RoleWebApi.Infrastructure.CrossCutting.DTO
{
    using System.Collections.Generic;

    public class ClientPagedList
    {
        public int IndexFrom { get; set; }
        public int PageIndex { get; set; }
        public int PageSize { get; set; }
        public int TotalCount { get; set; }
        public int TotalPages { get; set; }
        public IEnumerable<ClientDTO> Items { get; set; }
    }
}
